sdk/contracts/PerpsV2MarketInternalV2.ts

```diff
@@ -404,6 +418,48 @@ class FuturesMarketInternal {
 		return result.lt(0) ? BigNumber.from(0) : result;
 	};
 
+	_exactLiquidationPrice = async (
+		position: Position,
+		currentPrice: BigNumber,
+		approxLiquidationPrice: BigNumber
+	) => {
+		if (position.size.isZero()) {
+			return BigNumber.from('0');
+		}
+
+		const prices = Array.from(Array(1001).keys()).map((x) =>
+			wei(approxLiquidationPrice)
+				.mul(1 + (x - 500) / 1000)
+				.toBN()
+		);
+
+		// start with initial margin
+		const margins = await Promise.all(
+			prices.map(async (price) => await this._marginPlusProfitFunding(position, price))
```

Comment:
I think for this one we can remove the async from the function and then remove the awaits here